pip_compile/Tiltfile

```diff
@@ -0,0 +1,39 @@
+load("ext://uibutton", "cmd_button")
+
+
+SCRIPT = os.path.join(os.getcwd(), "scripts", "controller.py")
+DEFAULT_COMPILE_ARGS = ["--allow-unsafe", "--generate-hashes"]
+
+
+def pip_compile_button(
+    image,
+    target = "",
+    reqs_path = ".",
+    compile_args = DEFAULT_COMPILE_ARGS,
+):
+    """Add `pip-compile` button to all Tilt resources using `image`.
+
+    This button operates very similar to the `cancel` button example in tilt-extensions:
+    https://github.com/tilt-dev/tilt-extensions/blob/master/cancel/README.md
+    """
+    # Resolve absolute path for requirements directory.
+    reqs_path = os.path.realpath(reqs_path)
+
+    # Start a controller to watch for image changes and update related resources.
+    cmd = ["python", SCRIPT, image, "--target", target, "--reqs-path", reqs_path]
+    for arg in compile_args:
+        cmd.append("--compile-args=%s" % arg)
+
+    controller_resource = "pip-compile-controller:%s" % image
+    local_resource(
+        name = controller_resource,
```

Comment:
An image name could include `/`, but Tilt doesn't like that:

```
Loading Tiltfile at: /Users/noelle/dev/DirectEmployers/postajob/Tiltfile
Traceback (most recent call last):
  /Users/noelle/dev/DirectEmployers/postajob/Tiltfile:65:19: in <toplevel>
  /Users/noelle/dev/DirectEmployers/tilt-extensions/pip_compile/Tiltfile:28:19: in pip_compile_button
Error in local_resource: local_resource: for parameter "name": invalid value "pip-compile-controller:postajob/foo/bar": may not contain '/'
```

Can you tweak `controller_resource` to make it Tilt-compatible in that situation?

Reply:
Good find!!